maskit/circuits.py

```diff
@@ -1,12 +1,46 @@
-from maskit._masks import PerturbationAxis as Axis, DropoutMask
-from maskit._masked_circuits import MaskedCircuit
 import pennylane as qml
 from pennylane import numpy as np
+from typing import List, Tuple
+
+from maskit._masks import PerturbationAxis as Axis, DropoutMask
+from maskit._masked_circuits import MaskedCircuit
+from maskit.utils import cross_entropy
+
+
+def cost(
+    circuit,
+    params,
+    rotations: List,
+    masked_circuit: MaskedCircuit,
+):
+    return 1 - circuit(params, rotations, masked_circuit)[0]
+
+
+def cost_basis(
+    circuit,
+    params,
+    data,
+    target,
+    rotations: List,
+    masked_circuit: MaskedCircuit,
+    wires: int,
+    wires_to_measure: Tuple[int, ...],
+    interpret: Tuple[int, ...],
+):
+    prediction = circuit(
+        params, data, rotations, masked_circuit, wires, wires_to_measure
+    )
+    return cross_entropy(
+        predictions=np.array(
+            [elem for index, elem in enumerate(prediction) if index in interpret]
+        ),
```

Comment:
If I understand correctly, you can directly index here – assuming it's an array-like.
```suggestion
        predictions=prediction[interpret,],
```